### PR TITLE
Ports Slapping and Kissing

### PR DESCRIPTION
**Pull request by Fox-McCloud**
Ports TG's method of slapping and adds kiss blowing.

Instead of slapping being a mere emote that has text with it, or you have to input someone's name, precisely (with some weird names being oddly immune to input parameters), this transitions them to an object that gets put in your hand.

This generally makes things a bit more interact as you have to actually click on the person instead of just running the emote and that's that.

This also implements TG's kiss blowing emote.

When you use the emote you get a little heart icon in your hand; wherever you click next, the kiss will travel across the screen until it hits someone or hits a way.

Kisses deal no damage (though, if/when quirks are in, anxious people won't know how to react to them) and just have unique sound.

If you have the trait `KISS_OF_DEATH` then your kisses will uh....be a lot more deadly---you'll be a heartbreaker, quite literally; not only do they deal damage, but they straight up deal massive damage to someone's heart; enough to instantly kill it. I was going to add the lipstick of death, but it'll cause a lot of conflict with the mutations refactor, so I'll wait on that once this and the other are merged in.

Provides players with an additional little harmless interaction while also making an existing one a bit more interactive and immersive.

:cl: Fox McCloud
tweak: Slapping now gives you a slapping hand that you actually can click on someone with; slapping has a new sound too!
tweak: You can also slap a table with the slapping hand; regular intent allows you to do up to 3 rapid slaps that are quieter, or you can switch to harm intent and do one really big table slap that's louder
add: Adds in kiss blowing; use the kiss emote to activate it, then click where you want to blow the kiss to
/:cl:

### CODE REVIEW

File: code/game/objects/items/hand_item.dm [hunk 1]
```diff
@@ -0,0 +1,87 @@
+/obj/item/slapper
+	name = "slapper"
+	desc = "This is how real men fight."
+	icon_state = "latexballon"
+	item_state = ""
```
Comment (Fox-McCloud) [inline code/game/objects/items/hand_item.dm:5]: this has to be this way; if it's `null`, it'll inherit the `icon_state` as the default `item_state`

**Comment by MRgaty** [comment]
New ways to taunt prisoners as seccie, dope!

**Review by dearmochi — APPROVED** [review]
Two minor things:
- Nothing spawns if you do one of the emotes while handcuffed
- The kiss projectile travels really fast -- it feels less like you're hit by a romantic kiss and more by a truck, but that's just me!

Code is fine otherwise.

**Comment by Fox-McCloud** [comment]
>Nothing spawns if you do one of the emotes while handcuffed

That's intentional; nothing can be put in your hands, since, technically, you're technically handcuffed.

**Comment by dearmochi** [comment]
> 
> 
> > Nothing spawns if you do one of the emotes while handcuffed
> 
> That's intentional; nothing can be put in your hands, since, technically, you're technically handcuffed.

Mhm, I meant more like the `You ready your ... hand.` appears despite nothing spawning, which may cause some confusion

**Comment by Fox-McCloud** [comment]
>appears despite nothing spawning, which may cause some confusion

Fun. This means inventory code is handling something improperly--it's resulting in the object being dropped, apparently, but it doesn't check for if you can even hold something first then. Joy.

**Review by Dumbdumn5 — APPROVED** [review]
When these issues are fixed, hell yeah! :+1:

**Comment by Fox-McCloud** [comment]
Merge conflicts resolved.

I went ahead and implemented table slapping too.

File: code/game/objects/items/hand_item.dm [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+	blown_kiss.fire()
+	qdel(src)
+
+/obj/item/kisser/death
+	name = "kiss of death"
+	desc = "If looks could kill, they'd be this."
+	color = COLOR_BLACK
+	kiss_type = /obj/item/projectile/kiss/death
+
+/obj/item/projectile/kiss
```
Comment (AffectedArc07) [inline code/game/objects/items/hand_item.dm:77]: Well thats something you dont read every day

**Review by AffectedArc07 — APPROVED** [review]
If anyone starts being weird with this I am grinding your bones for a loaf of bread.

**Pull request merged**